src/gatsby-node.js

```diff
@@ -1,42 +1,47 @@
-const path = require('path');
-const { exists, writeFile, ensureDir } = require('fs-extra');
+const path = require('path')
+const fs = require('fs')
+const { promisify } = require('util')
 
-const getMetaRedirect = require('./getMetaRedirect');
+const stat = promisify(fs.stat)
+const writeFile = promisify(fs.writeFile)
+
+const { mkdirp } = require('fs-extra')
+
+const getMetaRedirect = require('./getMetaRedirect')
 
 async function writeRedirectsFile(redirects, folder, pathPrefix) {
-  if (!redirects.length) return;
+  if (!redirects.length) return
 
   for (const redirect of redirects) {
-    const { fromPath, toPath } = redirect;
+    const { fromPath, toPath } = redirect
+
+    const FILE_PATH = path.join(folder, fromPath.replace(pathPrefix, ''), 'index.html')
 
-    const FILE_PATH = path.join(
-      folder,
-      fromPath.replace(pathPrefix, ''),
-      'index.html'
-    );
+    const fileExists = await stat(FILE_PATH)
+      .then(stat => stat.isFile())
+      .catch(() => false)
 
-    const fileExists = await exists(FILE_PATH);
     if (!fileExists) {
-      try {
-        await ensureDir(path.dirname(FILE_PATH));
-      } catch (err) {
-        // ignore if the directory already exists;
-      }
-
-      const data = getMetaRedirect(toPath);
-      await writeFile(FILE_PATH, data);
+      const dirExists = await stat(path.dirname(FILE_PATH))
```

Comment:
fs-extra's `ensureDir` is already promisified and ensures that the directory exists. If the directory structure does not exist, it is created.

https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir.md#ensuredirdiroptionscallback